client/components/seo/meta-title-editor/index.jsx

```diff
@@ -51,81 +39,44 @@ const tokenMap = {
 	archives: [ 'siteName', 'tagline', 'date' ]
 };
 
-const tokenize = translate => value => {
-	if ( ! isString( value ) ) {
-		return value;
-	}
-
-	// find token key from translated label
-	// e.g. "Post Title" > postTitle: translate( 'Post Title' )
-	//         value          type           translation
-	const type = findKey( getValidTokens( translate ), matches( value ) );
-
-	return isUndefined( type )
-		? { type: 'string', isBorderless: true, value }
-		: { type, value };
+const getTokensForType = ( type, translate ) => {
+	return get( tokenMap, type, [] )
+				.reduce( ( allTokens, name ) => ( {
+					...allTokens,
+					[ name ]: get( getValidTokens( translate ), name, '' )
+				} ), {} );
 };
 
 export class MetaTitleEditor extends Component {
 	constructor( props ) {
 		super( props );
-
-		this.state = {
-			type: 'frontPage'
-		};
-
-		this.switchType = this.switchType.bind( this );
 		this.updateTitleFormat = this.updateTitleFormat.bind( this );
 	}
 
-	switchType( { value: type } ) {
-		this.setState( { type } );
-	}
-
-	updateTitleFormat( values ) {
-		const { onChange, translate, titleFormats } = this.props;
-		const { type } = this.state;
-
-		const tokens = removeBlanks( map( values, tokenize( translate ) ) );
+	updateTitleFormat( type, values ) {
+		const { onChange, titleFormats } = this.props;
 
 		onChange( {
 			...titleFormats,
-			[ type ]: tokens
+			[ type ]: values
 		} );
 	}
 
 	render() {
-		const { disabled, translate, titleFormats } = this.props;
-		const { type } = this.state;
-
-		const validTokens = getValidTokens( translate );
-
-		const values = map(
-			get( titleFormats, type, [] ),
-			token => 'string' !== token.type
-				? { ...token, value: validTokens[ token.type ] } // use translations of token names
-				: { ...token, isBorderless: true }               // and remove the styling on plain text
-		);
-
-		const suggestions = difference(
-			valuesOf( pick( validTokens, tokenMap[ type ] ) ), // grab list of translated tokens for this type
-			map( values, property( 'value' ) )                 // but remove tokens already in use in the format
-		);
+		const { disabled, titleFormats, translate } = this.props;
 
 		return (
 			<div className="meta-title-editor">
-				<SegmentedControl
-					initialSelected={ type }
-					options={ titleTypes( translate ) }
-					onSelect={ this.switchType }
-				/>
-				<TokenField
-					disabled={ disabled }
-					onChange={ this.updateTitleFormat }
-					saveTransform={ identity } // don't trim whitespace
-					suggestions={ suggestions }
-					value={ values }
-				/>
+				{ titleTypes( translate ).map( type =>
+					<TitleFormatEditor
+						key={ type.value }
+						disabled={ disabled }
+						onChange={ this.updateTitleFormat }
+						type={ disabled ? '' : type }
```

Comment:
This triggers a failed prop type warning because `type` is expected [to be an object](https://github.com/Automattic/wp-calypso/pull/7313/files#diff-dbf5a08e0e19cfb67fb595c9783359b8R287).  Any reason not to pass `type` always?